### PR TITLE
Make rust code compiling ;)

### PR DESCRIPTION
**Pull request by ignatenkobrain**

### CODE REVIEW

**Comment by ignatenkobrain** [comment]
I'm not sure if it works though :)

File: rust/src/main.rs [hunk 1]
```diff
@@ -33,108 +33,124 @@ pub struct EvTag {
...
+                    self.checksum_submodule(hash, &submodule)?;
+                }
+                ObjectType::Tag => {}
+                ObjectType::Any => panic!("Found an Any type?"),
+            }
+        }
+        Ok(())
+    }
+
+    fn checksum_commit_contents(
```
Comment (cgwalters) [inline rust/src/main.rs]: IMO this looks ugly...why did rustfmt do that?

**Comment by ignatenkobrain** [comment]
I think I broke it for submodules (or it didn't even work in rust version)..

File: rust/src/main.rs (hunk 1 above)
Comment (ignatenkobrain) [reply rust/src/main.rs]: I think because line is too long. Let me check RFC style

Comment (ignatenkobrain) [reply rust/src/main.rs]: https://github.com/rust-lang-nursery/rustfmt/blob/master/Configurations.md#function-arguments

**Comment by cgwalters** [comment]
> I think I broke it for submodules (or it didn't even work in rust version)..

I have a patch, will do after this.  Thanks!

**Comment by cgwalters** [comment]
Hmm wait you did this on the `rust` branch but that was old I think, the Rust bits have been in master for a while.

**Comment by cgwalters** [comment]
FWIW patch to make submodules work:

```
diff --git a/rust/src/main.rs b/rust/src/main.rs
index aedf818..26c3f6e 100644
--- a/rust/src/main.rs
+++ b/rust/src/main.rs
@@ -38,8 +38,8 @@ impl EvTag {
         })
     }
 
-    fn checksum_object(&self, object: &Object, hash: &mut Hasher) -> Result<(), Error> {
-        let odb = self.repo.odb()?;
+    fn checksum_object(&self, repo: &Repository, object: &Object, hash: &mut Hasher) -> Result<(), Error> {
+        let odb = repo.odb()?;
         let object = odb.read(object.id())?;
         let contentbuf = object.data();
         let header = format!("{} {}", object.kind().str(), contentbuf.len());
@@ -59,7 +59,7 @@ impl EvTag {
     }
 
     fn checksum_tree(&self, repo: &Repository, tree: Tree, hash: &mut Hasher) -> Result<(), Error> {
-        self.checksum_object(tree.as_object(), hash)?;
+        self.checksum_object(repo, tree.as_object(), hash)?;
         for entry in tree.iter() {
             match entry
                 .kind()
@@ -67,7 +67,7 @@ impl EvTag {
             {
                 ObjectType::Blob => {
                     let object = repo.find_object(entry.id(), entry.kind())?;
-                    self.checksum_object(&object, hash)?
+                    self.checksum_object(repo, &object, hash)?
                 }
                 ObjectType::Tree => {
                     let tree = repo.find_tree(entry.id())?;
@@ -90,7 +90,7 @@ impl EvTag {
         commit: Commit,
         hash: &mut Hasher,
     ) -> Result<(), Error> {
-        self.checksum_object(commit.as_object(), hash)?;
+        self.checksum_object(repo, commit.as_object(), hash)?;
         let tree = commit.tree()?;
         self.checksum_tree(repo, tree, hash)?;
         Ok(())
```

**Comment by ignatenkobrain** [comment]
@cgwalters oh, I didn't notice that master has rust stuff merged...

**Comment by ignatenkobrain** [comment]
@cgwalters but you should be able to do just git cherry-pick to master and it should apply just cleanly.

**Comment by cgwalters** [comment]
Yep, did that now, pushed to master.  I tweaked the commit messages a bit to have a `rust:` topic prefix and also added single-sentence rationale blurbs for most of them to satisfy my relatively high commit message standards.

Thanks a ton for the patches!

**Pull request merged**